analyzers/src/SonarAnalyzer.CSharp/Helpers/CSharpSymbolUsageCollector.cs

```diff
@@ -82,6 +88,12 @@ public override void Visit(SyntaxNode node)
                                             .Where(a => knownSymbolNames.Contains(a.AttributeClass.Name))
                                             .Select(a => a.AttributeClass));
                 }
+                else if (node.IsKind(SyntaxKindEx.PrimaryConstructorBaseType)
+                         && ((PrimaryConstructorBaseTypeSyntaxWrapper)node).Type.GetName() is var typeName
+                         && knownSymbolNames.Contains(typeName))
```

Comment:
Not reused, can be inlined
```suggestion
                         && knownSymbolNames.Contains(((PrimaryConstructorBaseTypeSyntaxWrapper)node).Type.GetName()))
```